pkg/cloud/awsprovider.go

```diff
@@ -774,6 +784,13 @@ func (aws *AWS) DownloadPricingData() error {
 
 	aws.ConfigureAuthWith(c) // load aws authentication from configuration or secret
 
+	// Need a valid value for at least one of these fields to consider spot pricing as enabled
+	if len(aws.SpotDataBucket) != 0 || len(aws.SpotDataRegion) != 0 || len(aws.ProjectID) != 0 {
+		aws.SpotRefreshEnabled = true
+	} else {
+		aws.SpotRefreshEnabled = false
+	}
```

Comment:
One question here about the lifecycle of this new value: what happens when a user starts with all empty fields and then fills in one or more fields. Do we update `aws.SpotRefreshEnabled` to go from `false` to `true`?

One alternative approach to consider, which would eliminate the need to update would be to make `aws.SpotRefreshEnabled` a function, like:
```go
func (aws) SpotRefreshEnabled() bool {
    return len(aws.SpotDataBucket) != 0 || len(aws.SpotDataRegion) != 0 || len(aws.ProjectID) != 0
}
```

Just a thought!

Reply:
To answer your question - modifying the spot configs in the UI and then clicking 'Save changes' in the Settings page should trigger the `DownloadPricingData()` method which recalculates the value of `aws.SpotRefreshEnabled` depending on the newly-modified configs.

Thanks for suggesting the function! This way I don't need to declare a new field in the aws struct, so I'll use that 👍

Reply:
Re: testing, I've added some logs and traced them in the container while trying out different config combinations in the frontend. The `pricingSourceStatus` endpoint returns the expected results in all scenarios. If the user declares these configs through their values.yaml file, they will also be accounted for when initialising the pods. Do you think there's any other edge case to look at?